src/admin/users/userDialogue.component.test.tsx

```diff
@@ -0,0 +1,186 @@
+import { screen } from '@testing-library/react';
+import userEvent, { UserEvent } from '@testing-library/user-event';
+import { MockInstance } from 'vitest';
+import { ogApi } from '../../api/api';
+import { renderComponentWithProviders } from '../../testUtils';
+import UserDialogue, { UserDialogueProps } from './userDialogue.component';
+
+describe('userDialogue', () => {
+  let props: UserDialogueProps;
+  let user: UserEvent;
+
+  const onClose = vi.fn();
+  const createView = () => {
+    return renderComponentWithProviders(<UserDialogue {...props} />);
+  };
+
+  beforeEach(() => {
+    props = { onClose: onClose, open: true, requestType: 'post' };
+    user = userEvent.setup();
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+  describe('add dialog', () => {
+    let axiosPostSpy: MockInstance;
+
+    beforeEach(() => {
+      axiosPostSpy = vi.spyOn(ogApi, 'post');
+    });
+    afterEach(() => {
+      vi.clearAllMocks();
+    });
+
+    it('renders the component correctly', async () => {
+      createView();
+      expect(screen.getByText('Add User')).toBeInTheDocument();
+    });
+
+    it('displays required error when username is empty', async () => {
+      createView();
+      const submitButton = screen.getByText('Submit');
+      await user.click(submitButton);
+
+      expect(
+        await screen.findByText('Username is required.')
+      ).toBeInTheDocument();
+    });
+
+    it('displays password field only when auth_type is "local"', async () => {
+      createView();
+
+      await user.type(screen.getByLabelText('Username'), 'new_user');
+      expect(screen.getByLabelText('Password')).toBeInTheDocument();
+    });
+
+    it('does not displays password field only when auth_type is "local"', async () => {
```

Comment:
I think this test is mean to be titled `does not display password field only when auth_type is "FedID"`